internal/factory/deployment.go

```diff
@@ -0,0 +1,50 @@
+package factory
+
+import (
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// NewDeployment stubs an instance of a deployment
+func NewDeployment(namespace, deploymentName, loggingComponent, component, impl string, podSpec core.PodSpec, visitors ...func(o runtime.Object)) *apps.Deployment {
+	selectors := map[string]string{
+		"provider":                        "openshift",
+		"component":                       component,
+		"logging-infra":                   loggingComponent,
+		constants.CollectorDeploymentKind: constants.DeploymentType,
+	}
+	labels := map[string]string{
+		"implementation": impl,
+	}
+
+	for k, v := range selectors {
+		labels[k] = v
+	}
+
+	annotations := map[string]string{
+		"target.workload.openshift.io/management": `{"effect": "PreferredDuringScheduling"}`,
+	}
+
+	strategy := apps.DeploymentStrategy{
+		Type: apps.RollingUpdateDeploymentStrategyType,
+		RollingUpdate: &apps.RollingUpdateDeployment{
+			MaxUnavailable: &intstr.IntOrString{
+				Type:   intstr.String,
+				StrVal: "100%",
```

Comment:
Having conversation the token does not immediately become unusable meaning we can have multiples running at the same time.  Kube docs say the default is 25% https://kubernetes.io/docs/concepts/workloads/controllers/deployment/ so I would try removing this value @wanghaoran1988 suggests we might start with 2 replicas